controllers/virtualmachine/volumebatch/volumebatch_controller.go

```diff
@@ -590,17 +598,90 @@ func (r *Reconciler) reconcileInstanceStoragePVCs(_ *pkgctx.VolumeContext) (bool
 	return true, nil
 }
 
-// handlePVCWithWFFC handles PVCs with WaitForFirstConsumer binding mode.
+// HandlePVCWithWFFC handles PVCs with WaitForFirstConsumer binding mode.
 // This ensures proper node selection for storage classes requiring it.
-func (r *Reconciler) handlePVCWithWFFC(_ *pkgctx.VolumeContext, _ vmopv1.VirtualMachineVolume) error {
-	// TODO: Implement WFFC (WaitForFirstConsumer) handling
-	// This method should:
-	// - Check storage class binding mode
-	// - Set selected node annotations for unbound PVCs
-	// - Handle node selection for proper PVC binding
-	// - Validate node availability and zone constraints
+func (r *Reconciler) HandlePVCWithWFFC(ctx *pkgctx.VolumeContext, volume vmopv1.VirtualMachineVolume) error {
+
+	if volume.PersistentVolumeClaim == nil || volume.PersistentVolumeClaim.InstanceVolumeClaim != nil {
+		return nil
+	}
+
+	pvc := corev1.PersistentVolumeClaim{}
+	pvcKey := client.ObjectKey{
+		Namespace: ctx.VM.Namespace,
+		Name:      volume.PersistentVolumeClaim.ClaimName,
+	}
+
+	if err := r.Get(ctx, pvcKey, &pvc); err != nil {
+		return fmt.Errorf("cannot get PVC: %w", err)
+	}
+
+	if pvc.Status.Phase == corev1.ClaimBound {
+		// Regardless of the StorageClass binding mode there is nothing to done if already bound.
+		return nil
+	}
+
+	if pvc.Annotations[constants.KubernetesSelectedNodeAnnotationKey] != "" {
```

Comment:
Why was this added? Unlike pods, the value of this is the zone name because of the `CNSSelectedNodeIsZoneAnnotationKey` annotation